Core/Interfaces/Security/Annotation/AnnotationsLoader.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ArtoxLab\Bundle\ClarcRbacBundle\Core\Interfaces\Security\Annotation;
+
+use Doctrine\Common\Annotations\Reader;
+use ReflectionClass;
+use ReflectionException;
+use RuntimeException;
+
+final class AnnotationsLoader implements AnnotationsLoaderInterface
+{
+    public function __construct(private Reader $reader)
+    {
+    }
+
+    /**
+     * @template T
+     *
+     * @param T $subject
+     *
+     * @return iterable<RulesVoter>
+     */
+    public function getAnnotations(object $subject): iterable
+    {
+        try {
+            $reflection = new ReflectionClass($subject);
+        } catch (ReflectionException) {
+            throw new RuntimeException('Failed to read annotation!');
+        }
+
+        yield from $this->lookup($reflection);
+    }
+
+    /**
+     * @return iterable<RulesVoter>
+     */
+    private function lookup(ReflectionClass $reflection): iterable
+    {
+        foreach ($reflection->getAttributes(RulesVoter::class) as $attribute) {
```

Comment:
`getAttributes` is available since PHP 8. Increase the required PHP version up to 8 or add an if statement
https://www.php.net/manual/ru/reflectionclass.getattributes.php